ts/src/gate.ts

```diff
@@ -5578,6 +5580,231 @@ export default class gate extends Exchange {
         return result;
     }
 
+    async fetchLedger (code: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name gate#fetchLedger
+         * @description fetch the history of changes, actions done by the user or operations that altered the balance of the user
+         * @see https://www.gate.io/docs/developers/apiv4/en/#query-account-book
+         * @see https://www.gate.io/docs/developers/apiv4/en/#list-margin-account-balance-change-history
+         * @see https://www.gate.io/docs/developers/apiv4/en/#query-account-book-2
+         * @see https://www.gate.io/docs/developers/apiv4/en/#query-account-book-3
+         * @see https://www.gate.io/docs/developers/apiv4/en/#list-account-changing-history
+         * @param {string|undefined} code unified currency code
+         * @param {int|undefined} since timestamp in ms of the earliest ledger entry
+         * @param {int|undefined} limit max number of ledger entries to return
+         * @param {object} params extra parameters specific to the gate api endpoint
+         * @returns {object} a [ledger structure]{@link https://docs.ccxt.com/#/?id=ledger-structure}
+         */
+        await this.loadMarkets ();
+        let type = undefined;
```

Comment:
humm something ain't right 
```
gate.fetchLedger()
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt12/ccxt/examples/py/cli.py", line 232, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/cjg/Git/ccxt12/ccxt/examples/py/cli.py", line 211, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt12/ccxt/python/ccxt/async_support/gate.py", line 5347, in fetch_ledger
    return self.parse_ledger(response, currency, since, limit)
  File "/Users/cjg/Git/ccxt12/ccxt/python/ccxt/async_support/base/exchange.py", line 1614, in parse_ledger
    itemOrItems = self.parse_ledger_entry(arrayData[i], currency)
  File "/Users/cjg/Git/ccxt12/ccxt/python/ccxt/async_support/gate.py", line 5414, in parse_ledger_entry
    'datetime': self.iso8601(timestamp),
  File "/Users/cjg/Git/ccxt12/ccxt/python/ccxt/base/exchange.py", line 1091, in iso8601
    utc = datetime.datetime.utcfromtimestamp(timestamp // 1000)
ValueError: year 55387 is out of range
```